test/code/plugins/VMInsightsDataCollector_subtest.rb

```diff
@@ -1141,13 +1144,14 @@ def <=>(o)
                 r = @dev <=> o.dev; return r unless r.zero?
                 r = @mp <=> o.mp; return r unless r.zero?
                 r = @size <=> o.size; return r unless r.zero?
-                @free <=> o.dev
+                r = @free <=> o.free; return r unless r.zero?
+                @type <=> o.type
```

Comment:
I'm confused how this could work. Isn't `o` always supposed to be an object created by the production code? If so, it isn't defined to have a `type` attribute.

Reply:
this o is not from production code. see how o contains dev, mp, size, type which are attributes of the Fs class defined in test code